instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/RocketMqConsumerTracer.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.rocketmq;
+
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER;
+import static io.opentelemetry.instrumentation.rocketmq.TextMapExtractAdapter.GETTER;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.List;
+import org.apache.rocketmq.common.message.MessageExt;
+
+public class RocketMqConsumerTracer extends BaseTracer {
+
+  private static final RocketMqConsumerTracer TRACER = new RocketMqConsumerTracer();
+
+  public static RocketMqConsumerTracer tracer() {
+    return TRACER;
+  }
+
+  @Override
+  protected String getInstrumentationName() {
+    return "io.opentelemetry.javaagent.rocketmq-client";
+  }
+
+  public Context startSpan(Context parentContext, List<MessageExt> msgs) {
+    MessageExt msg = msgs.get(0);
+    if (msgs.size() == 1) {
+      SpanBuilder spanBuilder = startSpanBuilder(msg).setParent(extractParent(msg));
+      return withClientSpan(parentContext, spanBuilder.startSpan());
+    } else {
+      SpanBuilder spanBuilder =
+          tracer
+              .spanBuilder(msg.getTopic() + " receive")
+              .setSpanKind(CONSUMER)
+              .setAttribute(SemanticAttributes.MESSAGING_SYSTEM, "rocketmq")
+              .setAttribute(SemanticAttributes.MESSAGING_OPERATION, "receive");
+      Context rootContext = withClientSpan(parentContext, spanBuilder.startSpan());
+      for (MessageExt message : msgs) {
+        createChildSpan(rootContext, message);
+      }
+      return rootContext;
+    }
+  }
+
+  public void createChildSpan(Context parentContext, MessageExt msg) {
+    SpanBuilder childSpanBuilder =
+        startSpanBuilder(msg)
+            .setParent(parentContext)
+            .addLink(Span.fromContext(extractParent(msg)).getSpanContext());
```

Comment:
In this particular case if the propagation is turned off you shouldn't add a link - it'll end up pointing to the `parentContext` anyway.

Reply:
I filed this spec issue we may be able to handle this in the SDK too https://github.com/open-telemetry/opentelemetry-specification/pull/1492

Reply:
Let's leave this code as is, next SDK will handle invalid links for us